proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -17,60 +24,238 @@
+                    final String testName = groupString.substring(0, bucketValueStart.get()).trim();
+                    final String bucketValueStr = groupString.substring(bucketValueStart.get()).trim();
+                    try {
+                        final Integer bucketValue = Integer.valueOf(bucketValueStr);
+                        builder.putForceGroup(testName, bucketValue);
+                        if (bucketAndPayloadValuesStr.length == FORCE_PARAMETER_MAX_SIZE && forcePayloadTests.contains(testName)) {
+                            final Payload payloadValue = parseForcePayloadString(bucketAndPayloadValuesStr[FORCE_PARAMETER_PAYLOAD_IDX]);
+                            if (payloadValue != null) {
+                                builder.putForcePayload(testName, payloadValue);
+                            }
+                        }
+                    } catch (final NumberFormatException e) {
+                        LOGGER.error("Unable to parse bucket value " + bucketValueStr + " as integer", e);
+                    }
+                }
+            }
+        }
+        return builder.build();
+    }
 
-            int bucketValueStart = piece.length() - 1;
-            for (; bucketValueStart >= 0; bucketValueStart--) {
-                if (!Character.isDigit(piece.charAt(bucketValueStart))) {
+    private static Optional<Integer> getBucketValueStart(final String groupString) {
+        int bucketValueStart = groupString.length() - 1;
+        for (; bucketValueStart >= 0; bucketValueStart--) {
+            if (!Character.isDigit(groupString.charAt(bucketValueStart))) {
+                break;
+            }
+        }
+        //  if no name or no value was found, it's not a valid proctor test bucket name
+        if ((bucketValueStart == groupString.length() - 1) || (bucketValueStart < 1)) {
+            return Optional.empty();
+        }
+        //  minus sign can only be at the beginning of a run
+        if (groupString.charAt(bucketValueStart) != '-') {
+            bucketValueStart++;
+        }
+
+        return Optional.of(bucketValueStart);
+    }
+
+    /**
+     * The format of the payloadString is the following:
+     * <ul>
+     *     <li>payloadType:payloadValue</li>
+     *     <li>Where payloadType is one of the 7 payload types supported by Proctor (stringValue, stringArray, doubleValue, longValue, longArray, map)</li>
+     *     <li>If payloadValue is an array is expected in the following format: [value,value,value]</li>
+     *     <li>If payloadValue is a map expecting following format: [key:value,key:value,key:value]</li>
+     * </ul>
+     */
+    @Nullable
+    public static Payload parseForcePayloadString(final String payloadString )
+    {
+        final Payload payload = new Payload();
+
+        final String[] payloadPieces = StringEscapeUtils.escapeJava(payloadString).split(":",2);
```

Comment:
Why do we need to call `StringEscapeUtils.escapeJava(...)` here? According to the Javadoc of the method, it will convert the string to one with java-style escaping (e.g., `stringValue:"foo"bar"` -> `stringValue:\"foo\"bar\"`).

Reply:
I guess several places that calls `substring(...)` will become simpler if this `escapeJava` can be removed.

Reply:
Good point after looking into it more agreed it is not needed and makes the rest of the code simpler and easier to understand. I come from a c/c++ background where string functions are less safe so mistakenly added this. Removed.